src/sonic-yang-models/yang-models/sonic-events-dhcp-relay.yang

```diff
@@ -0,0 +1,69 @@
+module sonic-events-dhcp-relay {
+    namespace "http://github.com/sonic-net/sonic-events-dhcp-relay";
+    yang-version 1.1;
+    prefix events-dhcp-relay;
+
+    import sonic-events-common {
+        prefix evtcmn;
+        revision-date 2022-12-01;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONIC dhcp-relay events";
+
+    revision 2022-12-01 {
+        description "dhcp-relay alert events.";
+    }
+
+    container sonic-events-dhcp-relay {
+        container dhcp-relay-discard {
+            evtcmn:ALARM_SEVERITY_MAJOR;
+
+            description "
+                Declares an event for dhcp-relay discarding packet on an
+                interface due to missing IP address assigned.
+                Params:
+                    name of the interface discarding.";
+
+            leaf ifname {
+                type string {
+                    pattern 'Ethernet([0-9]{1,3})';
+                }
```

Comment:
why do we use this pattern? 
sonic-port.yang does not have this limitation.

Reply:
When importing sonic-port.yang, I ran into issues when validating testcase with value of "Ethernet0". I believe the reason is because sonic-port is expecting a list.

Reply:
Changed to leafref